quick-start/src/main/ui/app/jobs/job-export.component.ts

```diff
@@ -1,36 +1,28 @@
-import {Component, HostListener, Inject, Input} from '@angular/core';
-import {MdlDialogReference, MdlDialogService} from '@angular-mdl/core';
+import {Component, Inject} from '@angular/core';
 import {JobService} from "./jobs.service";
+import {MdlDialogService} from '@angular-mdl/core';
```

Comment:
@alexsmr Can you take another look at this file?  This import is being used within the logic heavy export() function in this file to trigger a generic alert, not for spawning the dialog that the UI component lives in.

See Line 42 below:

```suggestion
            this.dialogService.alert("Unable to export jobs");
```

I don't see any way to get rid of this, since the logic for exporting the Jobs should stay within the Smart component.   Unless I'm misunderstanding your comment...

Reply:
Yes, I saw that. What I meant was moving the whole this.dialogService into UI and creating a reference and use ViewChild:

`@ViewChild(JobExportUiComponent)
  private exportUI: JobExportUiComponent;
`
And then you can call directly exportUI.alert() from parent logical component. Your exportUI.alert should delegate to this.dialogService.alert in UI component.

Reply:
Thanks for the guidance.  That's very clear.  The code has been updated to reflect your suggestion.